src/Planner/Planner.cpp

```diff
@@ -989,14 +1017,27 @@ void addWithFillStepIfNeeded(QueryPlan & query_plan,
 void addLimitByStep(
     QueryPlan & query_plan, const LimitByAnalysisResult & limit_by_analysis_result, const QueryNode & query_node, bool do_not_skip_offset)
 {
-    /// Constness of LIMIT BY limit is validated during query analysis stage
-    UInt64 limit_by_limit = query_node.getLimitByLimit()->as<ConstantNode &>().getValue().safeGet<UInt64>();
+    Field limit_by_limit_field = query_node.getLimitByLimit()->as<ConstantNode &>().getValue();
+    const Field converted_limit_value_uint = convertFieldToType(limit_by_limit_field, DataTypeUInt64());
+    if (converted_limit_value_uint.isNull())
+        throw Exception(
+            ErrorCodes::INVALID_LIMIT_EXPRESSION,
+            "The value {} of LIMIT in LIMIT BY expression is not representable as UInt64",
+            applyVisitor(FieldVisitorToString(), limit_by_limit_field));
+
```

Comment:
We should not have entered the function for negative or fractional limit. This function should only be entered for nonnegative integral limit. So, this check and exception seems redundant.

Please see the comment above around line 455.